test/runner.py

```diff
@@ -563,6 +567,94 @@ def cleanup_emscripten_temp():
         pass
 
 
+def print_repository_info(directory, repository_name):
+  import subprocess
+  current_commit = subprocess.run(['git', 'log', '-n1'], cwd=directory, stdout=subprocess.PIPE, text=True).stdout.strip()
+  print(f'\n{repository_name} {current_commit}\n')
+  local_changes = subprocess.run(['git', 'diff'], cwd=directory, stdout=subprocess.PIPE, text=True).stdout.strip()
+  if local_changes:
+    print(f'\n{local_changes}\n')
+
+
+def log_test_environment():
+  """Print detailed information about the current test environment. Useful for
+  logging test run configuration in a CI."""
+  print('======================== Test Setup ========================')
+  print(f'Test time: {datetime.datetime.now(datetime.timezone.utc).strftime("%A, %B %d, %Y %H:%M:%S %Z")}')
+  print(f'Python: "{sys.executable}". Version: {sys.version}')
+  print(f'Emscripten test runner path: "{os.path.realpath(__file__)}"')
+
+  if os.path.isdir(os.path.join(config.EMSCRIPTEN_ROOT, '.git')):
+    print(f'\nEmscripten repository: "{config.EMSCRIPTEN_ROOT}"')
+
+  emscripten_version = os.path.realpath(os.path.join(config.EMSCRIPTEN_ROOT, 'emscripten-version.txt'))
+  if os.path.isfile(emscripten_version):
+    print(f'emscripten-version.txt: {utils.read_file(emscripten_version).strip()}')
+
+  if os.path.isdir(os.path.join(config.EMSCRIPTEN_ROOT, '.git')):
+    print_repository_info(config.EMSCRIPTEN_ROOT, 'Emscripten')
+
+  print(f'EM_CONFIG: "{config.EM_CONFIG}"')
+  if os.path.isfile(config.EM_CONFIG):
+    print(f'\n{utils.read_file(config.EM_CONFIG).strip()}\n')
+
+  node_js_version = subprocess.run(config.NODE_JS + ['--version'], stdout=subprocess.PIPE, text=True).stdout.strip()
+  print(f'NODE_JS: {config.NODE_JS}. Version: {node_js_version}')
+
+  print(f'BINARYEN_ROOT: {config.BINARYEN_ROOT}')
+  wasm_opt_version = subprocess.run([os.path.join(config.BINARYEN_ROOT, 'bin', 'wasm-opt.exe'), '--version'], stdout=subprocess.PIPE, text=True).stdout.strip()
+  print(f'wasm-opt version: {wasm_opt_version}')
```

Comment:
We have `building.check_binaryen()` that does this so we could maybe refactor that to export `building.get_binaryen_version()`?